cosmwasm/contracts/wormchain-accounting/src/contract.rs

```diff
@@ -117,20 +117,28 @@ fn submit_observations(
     let observations: Vec<Observation> =
         from_binary(&observations).context("failed to parse `Observations`")?;
 
-    let events = observations
-        .into_iter()
-        .map(|o| {
-            let key = transfer::Key::new(o.emitter_chain, o.emitter_address.into(), o.sequence);
+    let mut details = Vec::with_capacity(observations.len());
+    let mut events = Vec::with_capacity(observations.len());
+    for o in observations {
+        let key = transfer::Key::new(o.emitter_chain, o.emitter_address.into(), o.sequence);
+        let (status, event) =
             handle_observation(deps.branch(), o, guardian_set_index, quorum, signature)
-                .with_context(|| format!("failed to handle observation for key {key}"))
-        })
-        .filter_map(Result::transpose)
-        .collect::<anyhow::Result<Vec<_>>>()
-        .context("failed to handle `Observation`")?;
+                .with_context(|| format!("failed to handle observation for key {key}"))?;
```

Comment:
If there's an error with an observation, should we emit an error event or something instead of returning via `?` operator?  I'm not sure how likely it is, but if it happens, it will block a guardian from submitting all observations I think.

Reply:
Now that duplicate signatures are not treated as errors, any errors we do encounter would be more serious ("insufficient balance", "digest mismatch", etc) so I don't think it's unreasonable to fail the entire batch.

That said, I'm also ok if we want to add a `TransferStatus::Error` variant so that we can continue processing the rest of the batch.

@bruce-riley @hendrikhofstadt thoughts?

Reply:
I've changed this so that we return an error status and emit an event whenever we fail to handle the observation.  This will not fail the other observations in the batch.